### PR TITLE
feat: Added KeyVault key & secret outputs

### LINKED ISSUE #3245: [AVM Module Issue]: add more outputs to the key submodule
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### Module Name

avm/res/key-vault/vault

### (Optional) Module Version

_No response_

### Description

Would be very useful to output the key full uri (eg: to provide it to the AzureSQL server for encryption)
basically add this to the main.bicep:

@description('The uri with version of the key.')
output keyUriWithVersion string = key.properties.keyUriWithVersion

### (Optional) Correlation Id

_No response_

### PR DESCRIPTION
**Pull request by AlexanderSehr**
## Description

- Added KeyVault key & secrets outputs (Note: An output map was technically not possibe)
- Moved UDTs to common-types module where possible
- Based on #3246
- Depends on #3606
- Depends on https://github.com/microsoft/PSRule/issues/2591 & Release https://github.com/Azure/PSRule.Rules.Azure/milestone/127

## Pipeline Reference

| Pipeline |
| -------- |
|    [avm.res.key-vault.vault](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.key-vault.vault.yml)      |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

### CODE REVIEW

**Comment by jtracey93** [comment]
Hey @AlexanderSehr ,

Firstly, thanks for your work on this PR!

We have made some changes to the AVM CI, detailed below, which means we need you to update your fork to pull in these latest changes and re-run your tests to show they still are passing prior to approving and merging this PR, as we don't and it fails once merged the publishing of your module will fail and will be blocked going forward until the test pass again via additional PRs.

### Changes to CI That Have Been Made That You Need To Take Action On

- https://github.com/Azure/bicep-registry-modules/pull/3745
    - Adds WAF Security PSRule Checks using this custom baseline of rules ([`/avm/utilities/pipelines/staticValidation/psrule/.ps-rule/cb-waf-security.Rule.yaml`](https://github.com/Azure/bicep-registry-modules/blob/main/avm/utilities/pipelines/staticValidation/psrule/.ps-rule/cb-waf-security.Rule.yaml))
    - **Action Required By You:**
        - Ingest/pull-in these changes into your PR branch and re-run your tests to ensure they pass
            - If any issues with the rules and you think it should apply, please ping/reach out to the core team

Any questions reach out to the AVM Core Team by tagging us in your PR here or internally via Teams

Thanks

Jack (AVM Core Team)

**Review by fblix — APPROVED** [review]

**Pull request merged**
